Configure isolated travis build for kohana/core

acoulton opened the pull request:
This is the first of the changes to add separate CI support for each module in isolation as discussed in kohana/kohana#50.

Once this is done and working across all the 3.3 modules, I'll merge up to 3.4 and make any required changes for that.

@shadowhand, @zombor or @Zeelot I think only you (as admins) can activate the travis hooks? Could you do that now and then I'll force a push to get the build running? I think you could activate for all the core modules now, obviously all the others will fail for the minute as they don't have travis configuration but I should get to them pretty quickly.

acoulton committed:
Configure standalone travis build for kohana/core

Adds and configures kohana/koharness to build a skeleton Kohana
application and then uses that environment to run the unit tests
on travis (and locally, if required).

acoulton committed:
Remove physical path dependency in Kohana::module tests

Backports dd926956 to 3.3 - MODPATH is unreliable in a koharness
environment as it will usually be a symlink. With this change the test
of Kohana::modules will pass wherever the code happens to be
deployed on the disk. This will conflict when we merge up to 3.4
because that commit uses short array syntax which is not available
in the 3.3 series.

lenton commented:
When I tried this, composer errored when resolving dependencies because kohana/koharness requires kohana/core.

acoulton commented:
@lenton that's weird, it's not doing that for me - I've just tried again on a clean clone and with the latest composer and it works fine.

`composer install --prefer-dist` gives me:

```
Loading composer repositories with package information
Installing dependencies (including require-dev)
  - Installing composer/installers (v1.0.18)
    Downloading: 100%

  - Installing symfony/yaml (v2.5.4)
    Downloading: 100%

  - Installing phpunit/php-text-template (1.2.0)
    Downloading: 100%

  - Installing phpunit/phpunit-mock-objects (1.2.3)
    Downloading: 100%

  - Installing phpunit/php-timer (1.0.5)
    Downloading: 100%

  - Installing phpunit/php-token-stream (1.2.2)
    Downloading: 100%

  - Installing phpunit/php-file-iterator (1.3.4)
    Downloading: 100%

  - Installing phpunit/php-code-coverage (1.2.18)
    Downloading: 100%

  - Installing phpunit/phpunit (3.7.37)
    Downloading: 100%

  - Installing kohana/unittest (dev-3.3/develop 45f215d)
    Downloading: 100%

  - Installing kohana/koharness (dev-master 67e705b)
    Downloading: 100%

phpunit/php-code-coverage suggests installing ext-xdebug (>=2.0.5)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
Writing lock file
Generating autoload files
```

Have also tried just `composer install` and I get the same commits but with the kohana ones as clones.

That said, koharness probably shouldn't require core - I think it only does because it has a couple of test helper classes that extend core ones but possibly they shouldn't be there anyway and even if they do I think it's OK for it to be the developer's issue if you try to use them in a project that doesn't otherwise have core present.

acoulton commented:
@shadowhand bump - could you activate the travis hooks for all the kohana modules? Should just need to click the relevant toggles in travis.

acoulton committed:
Specify the COMPOSER_ROOT_VERSION for travis installation

This works around the circular dependency on kohana/unittest > kohana/core
which leads to resolution problems when working on a feature branch.

acoulton committed:
Remove IRC notifications from travis

acoulton commented:
woot woot - this is now green, can someone review and merge? @lenton I've fixed the circular dependency issue with koharness but there's also one with unittest that may require a workaround for this repo (documented in the new readme) when working on a feature branch.

enov commented on .gitignore:
```diff
@@ -1,2 +1,5 @@
 *~
 *.swp
+/composer.lock
+/vendor
```
I _late_ noticed that the others have a \* like in `vendor/*`

acoulton replied:
Fixed in 526fe4d144 thanks for spotting.

enov commented:
These isolated Travis CI tests are huge improvements for Kohana. Thank you @acoulton !

How about some [![Build Status](https://travis-ci.org/kohana/core.svg?branch=3.3%2Fdevelop&)](https://travis-ci.org/kohana/core) Travis CI badges  in README files to celebrate? :smile:

acoulton commented:
@enov no worries - I thought of Travis badges (and indeed adding README to the modules that don't have one). What branches should we show though? Just 3.3/master and (when it comes) 3.4/master, or develop branches too?

enov commented:
I think badges should go on all branches that have tests configured. Why exclude branches?
